playgrounds/eav-checkbox-playground.R

```diff
@@ -0,0 +1,559 @@
+# token      <- "831D17A42D03BC448A7B1A08CDE19AD9" # longitudinal
+token      <- "B77827EB144D9B7D74F01E932C9F0B29" # neither
+
+fields  <- NULL
+# fields   <- c(
+#   # "record_id",
+#   "form_1_check_001"
+# )
+forms   <- NULL
+# forms   <- c(
+#   "form_1",
+#   # "form_2",
+#   "form_3"
+#   # "form_4"
+# )
+events  <- NULL
+# events  <- c(
+#   "event_1_arm_1"
+#   # "event_3_arm_1"
+# )
+records <- NULL
+
+blank_for_gray_form_status <- TRUE
+.complete_value_for_untouched_forms <- dplyr::if_else(
+  blank_for_gray_form_status,
+  NA_character_,
+  as.character(REDCapR::constant("form_incomplete"))
+)
+
+# NOTE: I think that an additional function parameter to specify how to handle
+#   untouched checkbox items might make sense. But we can also make it dependent
+#   on `blank_for_gray_form_status` if you prefer that. I just think some people
+#   might want to make different selections for either.
+blank_for_untouched_checkbox <- TRUE
+
+# ---- load-data ---------------------------------------------------------------
+# View(REDCapR::redcap_variables(redcap_uri, token)$data)
+system.time({
+  meta <- REDCapR:::redcap_metadata_internal(
+    redcap_uri,
+    token
+  )
+
+  ds_metadata          <- meta$d_variable
+  ds_metadata_plumbing <- ds_metadata |>
+    dplyr::filter(
+      plumbing
+    )
+
+  # only include selected fields/forms
+  if (!is.null(fields) & !is.null(forms)) {
+    ds_metadata <- ds_metadata |>
+      dplyr::filter(
+        field_name_base %in% fields | form_name %in% forms
+      )
+  } else if (!is.null(fields)) {
+    ds_metadata <- ds_metadata |>
+      dplyr::filter(
+        field_name_base %in% fields
+      )
+  } else if (!is.null(forms)) {
+    ds_metadata <- ds_metadata |>
+      dplyr::filter(
+        form_name %in% forms
+      )
+  }
+
+  col_types <- REDCapR::redcap_metadata_coltypes(
+    redcap_uri,
+    token,
+    print_col_types_to_console = FALSE
+  )
+
+  # only include selected fields/forms
+  col_types[[1]] <- col_types[[1]][ds_metadata$field_name]
+
+  ds_eav <- REDCapR:::redcap_read_eav_oneshot(
+    redcap_uri,
+    token,
+    fields               = fields,
+    forms                = forms,
+    events               = events,
+    records              = records,
+    export_survey_fields = TRUE
+  )$data
+})
+
+# ---- tweak-data --------------------------------------------------------------
+# NOTE: while "standard" exports always include `redcap_repeat_...` columns if
+#   the project has at least one repeating instrument, eav exports only include
+#   them IF
+#   - forms (or fields from forms) are included that are repeating
+#   - AND the event(s) for which they are set to repeating is/are included
+#   - AND there is actually data for the included participants
+#
+#   Thus we have to make this determination dependent on the data and cannot use
+#   meta$repeating
+.repeating    <- "redcap_repeat_instrument" %in% names(ds_eav)
+.longitudinal <- meta$longitudinal
+
```

Comment:
still not very clear the differences between `repeat` and `longitudinal`. Isn't both of them collecting the same forms multiple times?

Reply:
Yes, but 

- **longitudinal:** Is a more formal way of recollecting the same forms at different events. REDCap has a fixed event structure and forms are associated with events
- **repeating:** Is an additional construct that allows to repeat a given form *as many times as one wants to*, i.e., no fixed event structure and event-form association

In REDCap, the two can be combined, i.e., a project can be longitudinal with a strict form-event mapping but one or several of the forms can be set to repeating *within the event(s) that they are associated with*.

From a data perspective, the two are represented in the following way:

- **longitudinal:** adds a `redcap_event_name` column; every participant has one row per event, i.e., a "participant/event"
- **repeating:** adds two columns `redcap_repeat_instance` (a number 1 through n, indicating the iteration of a repeating instrument) and `redcap_repeat_instrument` (the form name of the repeating form); these lead to additional rows, one per repeating instance + repeating instrument

I would recommend playing around with the test projects I set up, that should make both the functionality in the GUI and the output data more clear.